enhancements/kubernetes-nmstate-integration.md

```diff
@@ -0,0 +1,165 @@
+---
+title: kubernetes-nmstate
+authors:
+  - "@schseba"
+reviewers:
+
+approvers:
+  - TBD
+  
+creation-date: 2019-12-18
+last-updated: 2019-12-18
+status: 
+---
+
+# kubernetes-nmstate
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+A proposal to deploy [kubernetes-nmstate](https://github.com/nmstate/kubernetes-nmstate/) on OpenShift.
+
+Node-networking configuration driven by Kubernetes and executed by
+[nmstate](https://nmstate.github.io/).
+
+## Motivation
+
+With hybrid clouds, node-networking setup is becoming even more challenging.
+Different payloads have different networking requirements, and not everything
+can be satisfied as overlays on top of the main interface of the node (e.g.
+SR-IOV, L2, other L2).
+The [Container Network Interface](https://github.com/containernetworking/cni)
+(CNI) standard enables different
+solutions for connecting networks on the node with pods. Some of them are
+[part of the standard](https://github.com/containernetworking/plugins), and there are
+others that extend support for [Open vSwitch bridges](https://github.com/kubevirt/ovs-cni),
+[SR-IOV](https://github.com/hustcat/sriov-cni), and more...
+
+However, in all of these cases, the node must have the networks setup before the
+pod is scheduled. Setting up the networks in a dynamic and heterogenous cluster,
+with dynamic networking requirements, is a challenge by itself - and this is
+what this project is addressing.
+
+### Goals
+
+- Deploy kubernetes-nmstate as part of openshift
```

Comment:
This isn't a goal. This is a solution. What's the high-level goal?

Reply:
I agree that this defines a solution and not a goal.  For those less familiar with nmstate, it would be good to explain why nmstate is the right solution to addressing a goal and if there are others evaluated.

Reply:
@bcrochet we could add a section that gives background about nmstate being a declarative interface to Enterprise Linux's network interface configuration tool of choice (NetworkManager).